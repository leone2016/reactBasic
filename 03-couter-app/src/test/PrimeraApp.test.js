import React from 'react';
/* const { render } = require("@testing-library/react");*/
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
const { PrimeraApp } = require("../PrimeraApp");

describe('UNIT TEST <PrimeraApp/>', () => {
    /*  test('should be return the message "Hello, I am Leo"', () => {
         const saludo = 'Hello, I am Leo';
         const { getByText } = render( <PrimeraApp saludo={saludo} />);
         expect( getByText(saludo)).toBeInTheDocument();
     })
      */

    test('should be show <PrimeraApp/> correctly', () => {
        const saludo = 'Hello, I am Leo';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should be show the subtitle sended for props', () => {

        const saludo = 'Hello, I am Leo';
        const subTitulo = 'soy un subtitulo';
        const wrapper = shallow(<PrimeraApp
            saludo={saludo}
            subTitulo={subTitulo} />);
            
        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subTitulo);
    })


})
