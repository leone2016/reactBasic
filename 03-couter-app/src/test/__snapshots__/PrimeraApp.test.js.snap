// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UNIT TEST <PrimeraApp/> should be show <PrimeraApp/> correctly 1`] = `
<Fragment>
  <h1>
    Hello, I am Leo
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
